runtime/autoload/provider/clipboard.vim

```diff
@@ -122,17 +129,23 @@ function! provider#clipboard#Executable() abort
     let s:paste['+'] = win32yank.' -o --lf'
     let s:copy['*'] = s:copy['+']
     let s:paste['*'] = s:paste['+']
-    return 'win32yank'
+    let name = 'win32yank'
   elseif exists('$TMUX') && executable('tmux')
     let s:copy['+'] = 'tmux load-buffer -'
     let s:paste['+'] = 'tmux save-buffer -'
     let s:copy['*'] = s:copy['+']
     let s:paste['*'] = s:paste['+']
-    return 'tmux'
+    let name = 'tmux'
+  else
+    let s:err = 'clipboard: No clipboard tool. :help clipboard'
+    return ''
   endif
 
-  let s:err = 'clipboard: No clipboard tool. :help clipboard'
-  return ''
+  let s:copy['+'] = s:split_cmd(get(s:copy, '+', v:null))
+  let s:paste['+'] = s:split_cmd(get(s:paste, '+', v:null))
+  let s:copy['*'] = s:split_cmd(get(s:copy, '*', v:null))
+  let s:paste['*'] = s:split_cmd(get(s:paste, '*', v:null))
```

Comment:
would prefer to update all the cases to set lists instead of setting them as strings and then immediately splitting them. that also eliminates the need for s:split_cmd(), the rework of the `return` statements, and having many `let s:foo` cases which are then immediately overwritten in the same scope.